File: ex08/go/kkw5240.go
package main

import (
	"fmt"
)

func main() {
	people, err := promptHowManyPeople()
	if err != nil {
		return
	}

	pizzas, err := promptHowManyPizzas()
	if err != nil {
		return
	}

	pieces, err := promptHowManyPieces()
	if err != nil {
		return
	}

	introducePizzaParty(people, pizzas)

	pizzaParty := &PizzaParty{
		people: people,
		pizzas: pizzas,
		pieces: pieces,
	}

	showCalcSharedPieces(pizzaParty)

	showLeftPieces(pizzaParty)
}

func promptHowManyPeople() (int, error) {
	fmt.Printf("How many people? ")

	var people int
	_, err := fmt.Scanln(&people)
	if err != nil {
		fmt.Printf(err.Error())
		return 0, err
	}

	return people, nil
}

func promptHowManyPizzas() (int, error) {
	fmt.Printf("How many pizzas do you have? ")

	var pizzas int
	_, err := fmt.Scanln(&pizzas)
	if err != nil {
		fmt.Printf(err.Error())
		return 0, err
	}

	return pizzas, nil
}

func promptHowManyPieces() (int, error) {
	fmt.Printf("How many pieces are in a pizza? ")

	var pieces int
	_, err := fmt.Scanln(&pieces)
	if err != nil {
		fmt.Printf(err.Error())
		return 0, err
	}

	return pieces, nil
}

func introducePizzaParty(people int, pizzas int) {
	fmt.Printf("%d people with %d pizzas\n", people, pizzas)
}

func showCalcSharedPieces(pizzaParty *PizzaParty) {
	fmt.Printf("Each person gets %d pieces of pizza.\n", pizzaParty.CalcPieces())
}

func showLeftPieces(pizzaParty *PizzaParty) {
	fmt.Printf("There are %d leftover pieces.\n", pizzaParty.CalcLeftover())
}

type PizzaParty struct {
	people int
	pizzas int
	pieces int
}

func (t *PizzaParty) CalcPieces() int {
	return t.pizzas * t.pieces / t.people
}

func (t *PizzaParty) CalcLeftover() int {
	return t.pizzas * t.pieces % t.people
}

File: ex08/java/kkw5240.java
package ex08.java;

import java.util.Scanner;

class Ex08 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.printf("How many people? ");
        int people = scanner.nextInt();

        System.out.printf("How many pizzas do you have? ");
        int pizzas = scanner.nextInt();

        System.out.printf("How many pieces are in a pizza? ");
        int pieces = scanner.nextInt();

        scanner.close();

        System.out.printf("\n%d people with %d pizzas.\n", people, pizzas);

        int sharedPieces = pizzas * pieces / people;
        System.out.printf("Each person gets %d pieces.\n", sharedPieces);

        int leftoverPieces = pizzas * pieces % people;
        System.out.printf("There are %d leftover pieces.\n", leftoverPieces);
    }
}